### ISSUE TITLE
[PUM 1.6.6] Accessibility: tabindex=”-1″ missing in templates/popup.php

### ISSUE THREAD

**Issue by Shemzone** [PopupMaker/Popup-Maker#469]
Hi there and out there
Not really help needed but a suggest.
To be compliant with accessibility concerns, you must add tabindex="-1" in templates/popup.php
First line should start with 
`<div tabindex="-1" id="pum-<?php pum_popup_ID(); ?>"...`
instead of 
`<div id="pum-<?php pum_popup_ID(); ?>"...`

Source: [https://webaim.org/techniques/keyboard/tabindex](https://webaim.org/techniques/keyboard/tabindex)
Or maybe have I missed something in the settings?

Regards

**Comment by danieliser** [comment]
Will have to test it. Not sure it would matter either way as we have already done extensive testing for accessibility over the last few years.

IE we wrote our own focus first element functionality and tab/focus lock when pop ups are open as well. 

They do mention modals but the part they don’t mention is that hidden elements can’t be focused anyways. Try tabbing into your pop ups content while it’s closed. Once you get to the last footer link it will tab back to the header. Once you open a Popup focus should be set to the first element that is focusable and trap tab key inside the Popup.

If this has no negative effect will see about adding it to the v1.7 beta this week.

**Comment by Shemzone** [comment]
Ok, thank you. 
In addition, after having checked (navigate with tab key), even with tabindex -1, the focus stayes under the popup. 
See at [http://www.tedxorleans.com/](http://www.tedxorleans.com/)
The focus is taken on the popup on mouseclick and then doesn't leave it until it's closed. 
The normal behavior should be the following: once the popup appears, the focus should be exclusively on the popup until its closing.

Regards

**Comment by danieliser** [comment]
@Shemzone Ahh I see. That actually got patched already (v1.6.7), was a typo in the JS events. Apparently it never got confirmed working so hasn't been officially released.

https://www.dropbox.com/s/htb566enpmv541u/popup-maker_v1.6.7-beta.2.zip?dl=0

That should make the tab/focus trapping work correctly.

**Comment by Shemzone** [comment]
Thank you @danieliser for this version. It works fine now.

**Comment by danieliser** [comment]
Awesome, newest version should work fine to then.
